ntbs-service/Pages/Alerts/TransferRequestAction.cshtml.cs

```diff
@@ -148,9 +148,9 @@ public async Task AcceptTransferAndDismissAlertAsync()
             Notification.HospitalDetails.TBServiceCode = TransferAlert.TbServiceCode;
             Notification.HospitalDetails.HospitalId = TargetHospitalId;
             Notification.HospitalDetails.CaseManagerId = TargetCaseManagerId;
-            await Service.UpdateHospitalDetailsAsync(Notification, Notification.HospitalDetails);
-            await _treatmentEventRepository.AddAsync(transferOutEvent);
-            await _treatmentEventRepository.AddAsync(transferInEvent);
+            Service.UpdateHospitalDetailsWithoutSave(Notification, Notification.HospitalDetails);
+            _treatmentEventRepository.AddWithoutSave(transferOutEvent);
+            _treatmentEventRepository.AddWithoutSave(transferInEvent);
             await _alertService.DismissAlertAsync(TransferAlert.AlertId, User.Username());

```

Comment:
I presume that the actual save is now going to come in here? Do we need it to? (i.e. is there anything in this method that we want to group with the earlier updates?)

I think we need to add an explicit save step, to make it clear what's going on (and to set the audit custom field). The easiest case is if we can put it before the call to the alert service - if it has to come in the middle then we'll potentially need to do something more clever.

Reply:
Yes, so at the moment the save is happening when SaveChanges is called inside DismissAlertAsync. I'm not sure what the best way to add the save step is, as this class doesn't have the context object. But if we are going to set the audit custom field here too then I can add a context field to this class and do both here.

Reply:
That would be reasonable. In this case, maybe we should make it an extension method on the context, rather than a class method here... semantically that would feel a bit better.

Another option would be to do it within the `ItemRepository` (in this case the `_treatmentEventRepository`). It's not exactly ideal because you're making changes in a couple of different services/repositories and doing the save unilaterally in one of them, but I think it's a reasonable approach too, and it saves injecting the context directly here.

Reply:
Went for the second approach